code/gui/serialbox.cpp

```diff
@@ -58,14 +58,32 @@ SerialBox::SerialBox(
     setup_power_slider(*(ui->left_slider));
     setup_power_slider(*(ui->right_slider));
     setup_slider(*(ui->delay_slider), Delay::DELAY_DEFAULT, Delay::DELAY_MIN, Delay::DELAY_MAX, Delay::DELAY_INTERVAL);
+    m_edit_boxes[0] = ui->up_box;
```

Comment:
use brace initialization in the constructor initialization list

`SerialBox() : ..., m_edit_boxes{ui->up_box, ...}` etc.

Reply:
I don't want to because it segfaults and I'm gonna cry because it makes no sense

Reply:
Yeah if I initialise them there then the array ends up being filled with null and/or garbage. I don't know why.

Reply:
oh true tm

Reply:
Goddamn it Jeff
Just baited me hard, I knew there was a reason I didn't do it like that the first time.

Reply:
I didn't realise the pointers weren't setup until setupUi lol I thought that just assigns the widgets